### ISSUE TITLE
VMR-Streamer cannot coexist with VBAN due to (SILENT) port conflict

### ISSUE THREAD

**Issue by codefaux** [vburel2018/Voicemeeter-SDK#3]
I'll keep the description short. Working with a streamer using two rigs, Gaming Rig receives VBAN Audio from Streaming Rig. Streaming Rig cannot run VMR-Streamer *and* VBAN due to listen port conflict.

Symptom:
- With VoiceMeeter Potato v3.0.2.1 and VMR-Streamer from same
- "Primary" machine runs VBAN, Control -and- Incoming Stream, accepting both from "Secondary" machine
- "Secondary" machine runs VBAN Outgoing Stream to "Primary" machine (Audio Stream established and working)
- "Secondary" machine runs VMR-Streamer, properly configured for VBAN Control stream to "Primary" machine
- -- VMR-Streamer fails to connect
- "Primary" machine shuts down VoiceMeeter Potato (or disables VBAN)
- "Primary" machine runs VMR-Streamer, same configuration
- -- VMR-Streamer connects and works properly
- "Primary" machine restarts VoiceMeeter Potato (or enables VBAN)
- -- VoiceMeeter engine throws error, unable to start VBAN, Socket In Use.

Conclusion; VMR-Streamer opens a *listen* socket on the same port VBAN uses for incoming connections, fails without indication of cause when this socket cannot open (Message implies VBAN *host* is at fault)

Suggested fix; VMR-Streamer EITHER:
- A) Does not need any listen port (since it is not *receiving* connections) and should AVOID OPENING such
- B) Should not need *the same* listen port number and should acquire port number via RANDOM FUNCTION or ALLOW CONFIGURATION

Possibly relevant code;
vban_cmd.c -- VBAN_InitSocket called by VBANCMD_StartThread

I can't get it to compile (very little Windows experience with C/C++ but plenty on the Linux side) or I would simply submit a PR.

**Comment by vburel2018** [comment]
Yes, of course, only one application at a time can listen to a given UDP port. The solution is to simply change the port on the VM-Streamer View... But you may want to keep distant Voicemeeter with Default UDP port for VBAN Stream Management/topology...

So, the solution we are studying is to allow Voicemeeter server to use a secondary UDP port (to have 2 Listening port).

**Comment by codefaux** [comment]
Okay -- *part* of my initial message may have come from a lack of understanding *why* Streamer View needs a listen port. Regardless, if you explore the details of my message, it's patently obvious that I know two things can't listen on the same port. In fact, I explained that as being the cause of the error. I called it a "port conflict". Please don't assume I'm an idiot, I spent a lot of time trying to communicate clearly with you.

As for "simply" changing the port on Streamer View, the only port I could see to change was where it's configured to connect to each VBAN instance, which also meant I had to change VBAN's port, which meant they were both on the same port again, so that's not even a little bit as helpful or simple as you might think.

Regarding a potential solution -- if you don't mind my inexperience making a suggestion -- Perhaps the RT Packet registration could be modified to allow a "target IP:Port" to be given, allowing Streamer View to listen on an arbitrarily user-configured port (or randomized) for its RT data, while still connecting to VoiceMeeter on its single server port? That seems easier than rigging a dual-server Listen in VBAN, but I don't know the architecture you're working with.


Skipping that part though, I'll state the other point more carefully:
Streamer View lies to the end user about *why* it fails, when it is unable to listen on the port because it's in use.
It claims "Voicemeeter is not responding on VBAN #1" -- it does NOT say, "Unable to bind socket - port in use"
Streamer View needs a "Port in use" error message, so users don't waste time attempting to chase down the wrong problem.

**Comment by vburel2018** [comment]
streamer view can now be configured with another listen port to receive RT-Packet from Voicemeeter. In other words, Voicemeeter now reply to some service request with the right port (sender port).

**Comment by codefaux** [comment]
Thank you for the attention.

I cannot test this change, as I abandoned Windows on my all of systems when Microsoft started pushing Windows 11.

I will close the issue, as you seem confident it is fixed.

Thank you for your time.
